### ISSUE TITLE
style prop isn't applied in v3

### ISSUE THREAD

**Issue by markphilpot** [Norserium/react-indiana-drag-scroll#75]
We needed to jump to v3 to take control over the style injection into shadow roots, but ran into this issue:

The `style` override prop is never applied to the ScrollContainer component

![CleanShot 2022-10-24 at 16 44 10@2x](https://user-images.githubusercontent.com/46191/197650764-a8b6ebb9-a46d-405d-9f3f-244b7c6afd32.jpg)

**Comment by Norserium** [comment]
@markphilpot, fixed in `3.0.1-alpha`. Try it.

**Comment by Norserium** [comment]
@markphilpot, any news?

**Comment by danijun96** [comment]
@Norserium I was trying to use onEndScroll prop but it doesn't exist as well. change to older version 2.2.0 to make it work

**Comment by Norserium** [comment]
@danijun96, it's fixed in `3.0.2-alpha`. Check [this out](https://codesandbox.io/s/react-indiana-drag-scroll-v-3-e8g0tq?file=/src/index.tsx).

**Comment by danijun96** [comment]
> @danijun96, it's fixed in `3.0.2-alpha`. Check [this out](https://codesandbox.io/s/react-indiana-drag-scroll-v-3-e8g0tq?file=/src/index.tsx).

now it works, but I will stick to 2.2.0 as this is alpha version and I am using this lib in prod. I also faced some other issue  
1. when I add css property `scroll-behavior: smooth;` to the container the drag stopped working the reason to add 'scroll-behavior: smooth;' is because I created two buttons next and prev to be able to move the scrollBar also smoothly.
Also thank you for the great work!

**Comment by Norserium** [comment]
> now it works, but I will stick to 2.2.0 as this is alpha version and I am using this lib in prod.

No problem, that's why it's called alpha.

> is because I created two buttons next and prev to be able to move the scrollBar also smoothly.

Did you try to set `behavior` in `scrollTo` options instead? [The example](https://codesandbox.io/s/react-indiana-drag-scroll-v-3-programmatic-scroll-w6vwud?file=/src/index.tsx).

**Comment by danijun96** [comment]
> Did you try to set `behavior` in `scrollTo` options instead? [The example](https://codesandbox.io/s/react-indiana-drag-scroll-v-3-programmatic-scroll-w6vwud?file=/src/index.tsx).

Yes something similar came to my mind as well but with `scrollBy` it also has this option for smooth behavior. But I faced a problem I will describe it below.
In my app when I hit the end of the scroll an API request is being called to get next elements, i am using onEndScroll to fetch this event and also have marked the last item to check when it's in view, so the problem was when the call was made i got rendered too many items instead of only what was needed, for example if 50 elements had to be rendered i got 910 rendered, not sure why this happens and it happens when moving the scroll bar with button. with dragging is okay. I am not sure if the problem is on my side, but with 2.2.0 works okay.

**Comment by Norserium** [comment]
Let me make this clear, you update to `3.0.2-alpha` and get this misbehavior immediately? 

Did you include the library styles?
```css
import 'react-indiana-drag-scroll/dist/style.css';
```

**Comment by danijun96** [comment]
> Let me make this clear, you update to `3.0.2-alpha` and get this misbehavior immediately?
> 
> Did you include the library styles?
> 
> ```css
> import 'react-indiana-drag-scroll/dist/style.css';
> ```

Yes I did all that.

**Comment by Norserium** [comment]
@danijun96, it's strange, but without debugging this case I couldn't help you.
